cohttp-eio/src/server.ml

```diff
@@ -116,14 +134,10 @@ let callback { conn_closed; handler } conn input output =
 let run ?max_connections ?additional_domains ?stop ~on_error socket server =
   Eio.Net.run_server socket ?max_connections ?additional_domains ?stop ~on_error
     (fun socket peer_address ->
-      try
-        Eio.Switch.run @@ fun sw ->
-        let () =
-          Logs.info (fun m ->
-              m "%a: accept connection" Eio.Net.Sockaddr.pp peer_address)
-        and input = Eio.Buf_read.of_flow ~max_size:max_int socket in
-        Eio.Buf_write.with_flow socket @@ fun output ->
-        callback server (sw, peer_address) input output
-      with Eio.Io (Eio.Net.E (Connection_reset _), _) ->
+      Eio.Switch.run @@ fun sw ->
+      let () =
         Logs.info (fun m ->
-            m "%a: disconnected" Eio.Net.Sockaddr.pp peer_address))
+            m "%a: accept connection" Eio.Net.Sockaddr.pp peer_address)
+      and input = Eio.Buf_read.of_flow ~max_size:max_int socket in
+      Eio.Buf_write.with_flow socket @@ fun output ->
```

Comment:
I would expect `with_flow` to raise if the connection is reset, so I think the error handling here is still needed.

Reply:
Thanks